Dependencies/xr/Source/ARCore/XR.cpp

```diff
@@ -642,6 +655,158 @@ namespace xr
+                    installTcs.complete();
+                }
+            })};
+
+            // Kick off the install request, and set the task for our caller to wait on.
+            CheckARCoreInstallStatus(true);
+            task = installTcs.as_task().then(runtimeScheduler, arcana::cancellation::none(), [resumeTicket = std::move(resumeTicket)](){});
+        }
+
+        return task;
+    }
+
+    arcana::task<void, std::exception_ptr> CheckCameraPermissionAsync(const Babylon::JsRuntimeScheduler& runtimeScheduler)
+    {
+        auto task = arcana::task_from_result<std::exception_ptr>();
+        
+        // Check if permissions are already granted.
+        if (!GetAppContext().checkSelfPermission(PERMISSION_NAME))
+        {
+            // Register for the permission callback request.
+            arcana::task_completion_source<void, std::exception_ptr> permissionTcs;
+            auto permissionTicket
+            {
+                AddRequestPermissionsResultCallback(
+                    [permissionTcs](int32_t requestCode, const std::vector<std::string>& permissionList, const std::vector<int32_t>& results) mutable
+                    {
+                        // Check if this is our permission request ID.
+                        if (requestCode == PERMISSION_REQUEST_ID)
+                        {
+                            // Loop over permission request results, and find the entry corresponding to the camera grant/deny.
+                            for (int i = 0; i < permissionList.size(); i++)
+                            {
+                                if (permissionList[i].find(PERMISSION_NAME) != std::string::npos)
+                                {
+                                    // If the permission is found and granted complete the task.
+                                    if (results[i] == 0 /* PackageManager.PERMISSION_GRANTED */)
+                                    {
+                                        permissionTcs.complete();
+                                        return;
+                                    }
+                                }
+                            }
+
+                            // Did not find permission in the list, or it was explicitly denied.  Complete the task with an error.
+                            std::ostringstream message;
+                            message << "Camera permission not acquired successfully";
+                            permissionTcs.complete(arcana::make_unexpected(make_exception_ptr(std::runtime_error{message.str()})));
+                        }
+                    })
+            };
+
+            // Kick off the permission check request, and set the task for our caller to wait on.
+            GetMainActivity().requestPermissions(PERMISSION_NAME, PERMISSION_REQUEST_ID);
+            task = permissionTcs.as_task().then(runtimeScheduler, arcana::cancellation::none(), [ticket = std::move(permissionTicket)](){});
```

Comment:
Same comment as above, I would expect you could use `inline_scheduler` here since the purpose is just to extend the lifetime of the ticket.